### PR TITLE
Speed up yolox training on small datasets

### PR DESCRIPTION
**Pull request by bognabylicka**
Enable yolox model to use `adaptive_repeat_times` to speed up training on small datasets

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
ote-test/1534/

**Comment by Ilya-Krylov** [comment]
run ote-test

File: external/mmdetection/detection_tasks/apis/detection/config_utils.py [hunk 1]
```diff
@@ -229,11 +226,10 @@ def prepare_work_dir(config: Config) -> str:
...
     # Save labels in data configs.
     for subset in ('train', 'val', 'test'):
-        cfg = config.data[subset]
-        if cfg.type == 'RepeatDataset' or cfg.type == 'MultiImageMixDataset':
-            cfg.dataset.labels = labels
-        else:
-            cfg.labels = labels
+        if subset == 'train':
+            cfg = get_data_train(config)
+        else: cfg = config.data[subset]
```
Comment (druzhkov-paul) [inline external/mmdetection/detection_tasks/apis/detection/config_utils.py]:
```suggestion
        else:
            cfg = config.data[subset]
```

File: external/mmdetection/detection_tasks/apis/detection/config_utils.py [hunk 2]
```diff
@@ -271,9 +267,9 @@ def patch_color_conversion(pipeline):
 
     assert 'data' in config
     for subset in ('train', 'val', 'test'):
-        cfg = config.data[subset]
-        if cfg.type == 'RepeatDataset' or cfg.type == 'MultiImageMixDataset':
-            cfg = cfg.dataset
+        if subset == 'train':
+            cfg = get_data_train(config)
+        else: cfg = config.data[subset]
```
Comment (druzhkov-paul) [inline external/mmdetection/detection_tasks/apis/detection/config_utils.py]: ditto

File: external/mmdetection/detection_tasks/apis/detection/config_utils.py [hunk 3]
```diff
@@ -331,3 +327,10 @@ def cluster_anchors(config: Config, dataset: DatasetEntity, model: BaseDetector)
     config.model.bbox_head.anchor_generator = config_generator
     model.bbox_head.anchor_generator = model_generator
     return config, model
+
+
+def get_data_train(config):
```
Comment (druzhkov-paul) [inline external/mmdetection/detection_tasks/apis/detection/config_utils.py]: Please, add type hints.

**Review by druzhkov-paul — DISMISSED** [review]

**Comment by druzhkov-paul** [comment]
run ote-test

**Comment by druzhkov-paul** [comment]
run ote_sdk tests

**Pull request merged**